electrum/submarine_swaps.py

```diff
@@ -241,6 +296,87 @@ def add_lnwatcher_callback(self, swap: SwapData) -> None:
         callback = lambda: self._claim_swap(swap)
         self.lnwatcher.add_callback(swap.lockup_address, callback)
 
+    def hold_invoice_callback(self, payment_hash):
+        key = payment_hash.hex()
+        if key in self.swaps:
+            swap = self.swaps[key]
+            if swap.funding_txid is None:
+                coro = self.start_normal_swap(swap, None, None)
+                asyncio.run_coroutine_threadsafe(coro, self.network.asyncio_loop)
```

Comment:
note: re timing out hold-invoices, we need to consider how some existing timeouts interact with that, e.g.: https://github.com/spesmilo/electrum/blob/079c0ce5b2d7afc86876aa22ca8126b31b0ac6d1/electrum/lnutil.py#L416-L418

Reply:
Hold invoices are no longer timed out after they have been accepted (see 141cd524bc4881cee21dc5d265c4a80ffe4b033d)

However, there is still a possible interaction between CLTV expiration and swap expiration: 
 - alice initiates reverse swap (offers HTLC)
 - bob funds the swap
 - alice does not publish her claim transaction
 - the CLTV - 72 deadline is reached
 - alice publishes claim transaction (the swap has not expired yet)
 - bob will extract preimage and fulfill the HTLC. at that point, `should_be_closed_due_to_expiring_htlcs` will return `True`.  if bob's channel counterparty does not revoke quickly, I believe the channel might be force closed by bob.